spec/Section 3 -- Type System.md

```diff
@@ -1340,14 +1357,280 @@ Union type extensions have the potential to be invalid if incorrectly defined.
 
 1. The named type must already be defined and must be a Union type.
 2. The member types of a Union type extension must all be Object base types;
-   Scalar, Interface and Union types must not be member types of a Union.
-   Similarly, wrapping types must not be member types of a Union.
+   Scalar, Interface, Union and Tagged types must not be member types of a
+   Union.  Similarly, wrapping types must not be member types of a Union.
 3. All member types of a Union type extension must be unique.
 4. All member types of a Union type extension must not already be a member of
    the original Union type.
 5. Any non-repeatable directives provided must not already apply to the
    original Union type.
 
+
+## Tagged Types
+
+TaggedTypeDefinition : Description? tagged Name Directives[Const]? TaggedMembersDefinition?
+
+TaggedTypeExtension :
+  - extend tagged Name Directives[Const]? TaggedMembersDefinition
+  - extend tagged Name Directives[Const]
+
+TaggedMembersDefinition : { TaggedMemberDefinition+ }
+
+TaggedMemberDefinition : Description? Name : Type Directives[Const]?
+
+Tagged types represent a list of possible named members, exactly one of which
+must be present. This resulting member must yield a value of a specific type.
+In outputs, like for Objects, Tagged types should be serialized as ordered
```

Comment:
I don't see why ordering plays a role here, since there is only ever one field. Of course, the fields of the parent Object type in which the Tagged type is used must be ordered - but that is a concern of the Object type.

Reply:
(For others, it's because `__typename` and aliases can also exist.)